src/Microsoft.AspNetCore.Localization/CookieRequestCultureProvider.cs

```diff
@@ -36,6 +42,8 @@ public override Task<ProviderCultureResult> DetermineProviderCultureResult(HttpC
                 throw new ArgumentNullException(nameof(httpContext));
             }
 
+            _logger = httpContext.RequestServices.GetService<ILogger>() ?? NullLogger.Instance;
```

Comment:
Yup, that's sort of what I was asking @hishamco to do. We'd have a no-arg ctor for back-compat and one which accepts an `ILogger`. We'll call the new ctor via a new options setup type. The breaking change here would be that https://github.com/aspnet/Localization/blob/f650de8cdf5214bcbb2e88210baac0f99ce3d053/src/Microsoft.AspNetCore.Localization/RequestLocalizationOptions.cs#L24 wouldn't be initialized in the ctor, but from options setup. However if you go via the expected code path of pulling options from the DI container, everything should be hunky dory.